docs/src/main/asciidoc/tests-with-coverage-guide.adoc

```diff
@@ -0,0 +1,701 @@
+
+[source,java]
+----
+package org.acme.quickstart;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Tag;
+
+import java.util.UUID;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+
+@QuarkusTest
+@Tag("integration")
+public class GreetingResourceTest {
+
+    @Test
+    public void testHelloEndpoint() {
+        given()
+          .when().get("/hello")
+          .then()
+             .statusCode(200)  
+             .body(is("hello"));
+    }
+
+    @Test
+    public void testGreetingEndpoint() {
+        String uuid = UUID.randomUUID().toString();
+        given()
+          .pathParam("name", uuid)
+          .when().get("/hello/greeting/{name}")
+          .then()
+            .statusCode(200)
+            .body(is("hello " + uuid));
+    }
+}
+----
+
+One will be our example of a Unit Test and one will be our example of Integration Test.
+
+== Separating executions for Unit Tests and Integration Tests
+
+You may want to consider that JUnits and QuarkusTests are two different kind of tests and that they deserve to be separated. This way you could run them separately, in different cases or some more often than the others.
+In order to do so, we'll use a feature of JUnit 5 that allows us to tag some tests. Let's tag `GreetingResourceTest.java` and specify that it is an Integration Test:
+
+[source,java]
+----
+import org.junit.jupiter.api.Tag;
+...
+
+@QuarkusTest
+@Tag("integration")
+public class GreetingResourceTest {
+    ...
+}
+----
+
+We're now able to distinguish unit tests and integration tests. Now, let's bind them to different Maven lifecycle phases. Let's use surefire to bind unit tests to the *test* phase and the integration tests to the *integration-test* phase.
+
+[source,xml,subs=attributes+]
+----
+<project>
+    ...
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <version>${surefire-plugin.version}</version>
+                <configuration>
+                    <excludedGroups>integration</excludedGroups>
+                    <systemProperties>
+                        <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                    </systemProperties>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>integration-tests</id>
+                        <phase>integration-test</phase>
+                        <goals>
+                            <goal>test</goal>
+                        </goals>
+                        <configuration>
+                            <excludedGroups>!integration</excludedGroups>
+                            <groups>integration</groups>
+                        </configuration>
+                    </execution>
+                </executions>
```

Comment:
With your config the integration tests will be run twice: as part of the default-test and then as integration tests. Shouldn't it be something like
````
                <executions>
                    <execution>
                        <id>default-test</id>
                        <phase>test</phase>
                        <goals>
                            <goal>test</goal>
                        </goals>
                        <configuration>
                            <excludedGroups>integration</excludedGroups>
                        </configuration>
                    </execution>
                    <execution>
                        <id>integration-tests</id>
                        <phase>integration-test</phase>
                        <goals>
                            <goal>test</goal>
                        </goals>
                        <configuration>
                            <groups>integration</groups>
                        </configuration>
                    </execution>
                </executions>
````

Reply:
I may have missed something but it should not. By default, the tag "integration" is excluded. That is what applies during the test phase. Then during the integration-test phase, the configuration of integration-tests prevails and only ITs are ran.

Here are the logs of the run:
```bash
╰─ mvn clean verify
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------------< org.acme:tests-with-coverage >--------------------
[INFO] Building tests-with-coverage 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ tests-with-coverage ---
[INFO] Deleting /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ tests-with-coverage ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ tests-with-coverage ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/classes
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:instrument (instrument-ut) @ tests-with-coverage ---
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ tests-with-coverage ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ tests-with-coverage ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/test-classes
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ tests-with-coverage ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.acme.quickstart.GreetingServiceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.055 s - in org.acme.quickstart.GreetingServiceTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:restore-instrumented-classes (restore-ut) @ tests-with-coverage ---
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ tests-with-coverage ---
[INFO] Building jar: /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/tests-with-coverage-1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- quarkus-maven-plugin:0.19.1:build (default) @ tests-with-coverage ---
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Beginning quarkus augmentation
[INFO] [org.jboss.threads] JBoss Threads version 3.0.0.Beta4
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 1026ms
[INFO] [io.quarkus.creator.phase.runnerjar.RunnerJarPhase] Building jar: /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/tests-with-coverage-1.0-SNAPSHOT-runner.jar
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:instrument (instrument-it) @ tests-with-coverage ---
[INFO] 
[INFO] --- maven-surefire-plugin:2.22.1:test (integration-tests) @ tests-with-coverage ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.acme.quickstart.GreetingResourceTest
2019-07-22 19:18:31,260 INFO  [io.qua.dep.QuarkusAugmentor] (main) Beginning quarkus augmentation
2019-07-22 19:18:32,041 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 781ms
2019-07-22 19:18:32,487 INFO  [io.quarkus] (main) Quarkus 0.19.1 started in 0.437s. Listening on: http://[::]:8081
2019-07-22 19:18:32,487 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy]
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.146 s - in org.acme.quickstart.GreetingResourceTest
2019-07-22 19:18:33,479 INFO  [io.quarkus] (main) Quarkus stopped in 0.009s
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:restore-instrumented-classes (restore-it) @ tests-with-coverage ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:report (report-it) @ tests-with-coverage ---
[INFO] Loading execution data file /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/jacoco-it.exec
[INFO] Analyzed bundle 'tests-with-coverage' with 2 classes
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:report (report-ut) @ tests-with-coverage ---
[INFO] Loading execution data file /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/jacoco-ut.exec
[INFO] Analyzed bundle 'tests-with-coverage' with 2 classes
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:merge (merge-results) @ tests-with-coverage ---
[INFO] Loading execution data file /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/jacoco-ut.exec
[INFO] Loading execution data file /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/jacoco-it.exec
[INFO] Writing merged execution data to /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/jacoco.exec
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:report (post-merge-report) @ tests-with-coverage ---
[INFO] Loading execution data file /Users/legone/Documents/git_clones/quarkus-quickstarts/tests-with-coverage/target/jacoco.exec
[INFO] Analyzed bundle 'tests-with-coverage' with 2 classes
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11.081 s
[INFO] Finished at: 2019-07-22T19:18:34+02:00
[INFO] ------------------------------------------------------------------------
```

Reply:
Looks like you forgot to push your changes. Here is what I see with your config
````
[INFO] --- maven-surefire-plugin:3.0.0-M2:test (default-test) @ tests-with-coverage ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.acme.quickstart.GreetingResourceTest
2019-07-23 11:18:30,524 INFO  [io.qua.dep.QuarkusAugmentor] (main) Beginning quarkus augmentation
2019-07-23 11:18:31,287 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 763ms
2019-07-23 11:18:31,593 INFO  [io.quarkus] (main) Quarkus 0.19.1 started in 0.299s. Listening on: http://[::]:8081
2019-07-23 11:18:31,593 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy]
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.678 s - in org.acme.quickstart.GreetingResourceTest
[INFO] Running org.acme.quickstart.GreetingServiceTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in org.acme.quickstart.GreetingServiceTest
2019-07-23 11:18:32,539 INFO  [io.quarkus] (main) Quarkus stopped in 0.009s
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ tests-with-coverage ---
[INFO] Building jar: /home/aloubyansky/playground/new/target/tests-with-coverage-1.0-SNAPSHOT.jar
[INFO] 
[INFO] --- quarkus-maven-plugin:0.19.1:build (default) @ tests-with-coverage ---
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Beginning quarkus augmentation
[INFO] [org.jboss.threads] JBoss Threads version 3.0.0.Beta4
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 974ms
[INFO] [io.quarkus.creator.phase.runnerjar.RunnerJarPhase] Building jar: /home/aloubyansky/playground/new/target/tests-with-coverage-1.0-SNAPSHOT-runner.jar
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M2:test (integration-tests) @ tests-with-coverage ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.acme.quickstart.GreetingResourceTest
2019-07-23 11:18:35,527 INFO  [io.qua.dep.QuarkusAugmentor] (main) Beginning quarkus augmentation
2019-07-23 11:18:36,270 INFO  [io.qua.dep.QuarkusAugmentor] (main) Quarkus augmentation completed in 743ms
2019-07-23 11:18:36,554 INFO  [io.quarkus] (main) Quarkus 0.19.1 started in 0.273s. Listening on: http://[::]:8081
2019-07-23 11:18:36,554 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy]
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.066 s - in org.acme.quickstart.GreetingResourceTest
2019-07-23 11:18:37,516 INFO  [io.quarkus] (main) Quarkus stopped in 0.004s
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
````

Reply:
Hold on, perhaps, I misspelled something. As now I re-did it and it works.

Reply:
Ok, I'm resolving this one.